src/default-client/index.ts

```diff
@@ -0,0 +1,628 @@
+  }
+
+  public async clearCache(): Promise<void> {
+    try {
+      await Promise.all([
+        this._cache.dataPaths.clear(),
+        this._cache.dataEntities.clear(),
+        this._cache.responses.clear(),
+      ]);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async exportCaches(tag: any): Promise<ExportCachesResult> {
+    try {
+      return this._cache.export(tag);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async getDataEntityCacheEntry(key: string): Promise<ObjectMap | undefined> {
+    try {
+      return this._cache.dataEntities.get(key);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async getDataEntityCacheSize(): Promise<number> {
+    try {
+      return this._cache.dataEntities.size();
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async getDataPathCacheEntry(key: string): Promise<any> {
+    try {
+      return this._cache.dataPaths.get(key);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async getDataPathCacheSize(): Promise<number> {
+    try {
+      return this._cache.dataPaths.size();
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async getResponseCacheEntry(key: string): Promise<ResponseCacheEntryResult | undefined> {
+    try {
+      const entry = await this._cache.responses.get(key);
+      if (!entry) return undefined;
+      const { cacheMetadata, data } = entry;
+      return { cacheMetadata: createCacheMetadata({ cacheMetadata }), data };
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async getResponseCacheSize(): Promise<number> {
+    try {
+      return this._cache.responses.size();
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async importCaches(caches: ExportCachesResult): Promise<void> {
+    try {
+      return this._cache.import(caches);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  public async request(query: string, opts: RequestOptions = {}): Promise<RequestResult> {
+    let errors: Error[] = [];
+
+    if (!isString(query)) {
+      errors.push(new TypeError("Request expected query to be a string."));
+    }
+
+    if (!isPlainObject(opts)) {
+      errors.push(new TypeError("Request expected opts to be a plain object."));
+    }
+
+    if (errors.length) return Promise.reject(errors);
+
+    if (opts.fragments && (!isArray(opts.fragments) || opts.fragments.some((value) => !isString(value)))) {
+      return Promise.reject(new TypeError("Request expected opts.fragments to be an array of strings."));
+    }
+
+    try {
+      const context: RequestContext = { fieldTypeMap: new Map() };
+      const updated = await this._requestParser.updateRequest(query, opts, context);
+      errors = validate(this._schema, updated.ast) as GraphQLError[];

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4325)